adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -108,11 +108,30 @@ def __init__(
         if self._socknum == SOCKET_INVALID:
             raise RuntimeError("Failed to allocate socket.")
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.disconnect()
+        while self.status == adafruit_wiznet5k.SNSR_SOCK_FIN_WAIT:
```

Comment:
Want to add a retry loop here to avoid users getting stuck indefinitely and raise an error if we cant disconnect?